### PR TITLE
Add exercise and user example for using psyplot and iconarray

### PR DESCRIPTION
**Pull request by AnnikaLau**
I added the exercise [ex3_03_psyplot.ipynb](https://github.com/AnnikaLau/pyvis/blob/icon-vis/Part3/ex3_03_psyplot.ipynb) and renamed the other exercises so that it would appear at the correct place.

I also added the user example [ex4_4_cosmo_pollen_S_Adamov.ipynb](https://github.com/AnnikaLau/pyvis/blob/icon-vis/Part4/ex4_4_cosmo_pollen_S_Adamov.ipynb), which uses psyplot and iconarray.

For the exercises, I updated the environment.yml file to include all needed dependencies.

### CODE REVIEW

File: Part3/ex3_03_psyplot.ipynb [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+    "warnings.filterwarnings(\"ignore\", category=FutureWarning)\n",
+    "warnings.filterwarnings = lambda *a, **kw: None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ced48a8d",
+   "metadata": {},
+   "source": [
+    "# Exercise 3_psyplot Mesh plots (psyplot)\n",
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:23]:
```suggestion
    "# Exercise 3.3: Mesh plots (psyplot)\n",
```

File: Part3/ex3_03_psyplot.ipynb [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+   "execution_count": null,
+   "id": "e18d0f51",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ftp_pyvis = \"ftp://iacftp.ethz.ch/pub_read/alauber/pyvis/\"\n",
+    "filename = \"my_exp1_atm_3d_ml_20180921T000000Z_t2m.nc\"\n",
+    "urllib.request.urlretrieve(ftp_pyvis+filename, filename)\n",
+    "filename = \"rmp_my_exp1_atm_3d_ml_20180921T000000Z.nc\"\n",
+    "urllib.request.urlretrieve(ftp_pyvis+filename, filename)"
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:79]:
```suggestion
    "ftp_pyvis = \"ftp://iacftp.ethz.ch/pub_read/alauber/pyvis/\"\n",
    "filename = \"../data/my_exp1_atm_3d_ml_20180921T000000Z_t2m.nc\"\n",
    "urllib.request.urlretrieve(ftp_pyvis + filename, filename)\n",
    "filename = \"../data/rmp_my_exp1_atm_3d_ml_20180921T000000Z.nc\"\n",
    "urllib.request.urlretrieve(ftp_pyvis + filename, filename)"
```

File: Part3/ex3_03_psyplot.ipynb [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "55cab2c0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds = xr.open_dataset(\"my_exp1_atm_3d_ml_20180921T000000Z_t2m.nc\")\n",
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:97]:
```suggestion
    "ds = xr.open_dataset(\"../data/my_exp1_atm_3d_ml_20180921T000000Z_t2m.nc\")\n",
```

File: Part3/ex3_03_psyplot.ipynb [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f76604d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds_rmp = xr.open_dataset(\"rmp_my_exp1_atm_3d_ml_20180921T000000Z.nc\")\n",
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:142]:
```suggestion
    "ds_rmp = xr.open_dataset(\"../data/rmp_my_exp1_atm_3d_ml_20180921T000000Z.nc\")\n",
```

File: Part3/ex3_03_psyplot.ipynb [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d64a4948",
+   "metadata": {},
+   "source": [
+    "### Exercise\n",
+    " * Plot the 2m temperature over Switzerland\n",
+    " * Use the projection=ccrs.Robinson() (don't forget to transform the data)\n",
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:153]:
```suggestion
    " * Use `projection=ccrs.Robinson()` (don't forget to transform the data)\n",
```

File: Part3/ex3_03_psyplot.ipynb [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e62355b5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load dataset with psyplot\n",
+    "ds_icon = psy.open_dataset(\"my_exp1_atm_3d_ml_20180921T000000Z_t2m.nc\")\n",
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:247]:
```suggestion
    "ds_icon = psy.open_dataset(\"../data/my_exp1_atm_3d_ml_20180921T000000Z_t2m.nc\")\n",
```

File: Part3/ex3_03_psyplot.ipynb [hunk 7]
```diff
@@ -0,0 +1,471 @@
...
+   "id": "dac55aeb",
+   "metadata": {},
+   "source": [
+    "I guess we could do that nicer....\n",
+    "\n",
+    "Checkout the [available formatoptions](https://psyplot.github.io/psy-maps/generated/psyplot.project.plot.mapplot.html) and try to make the same plot as before.\n",
+    "\n",
+    "\n",
+    "### Exercise\n",
+    " Use plot_icon.update(..) to adapt the plot in the following way: \n",
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:280]:
```suggestion
    " Use `plot_icon.update(...)` to adapt the plot in the following way: \n",
```

File: Part3/ex3_03_psyplot.ipynb [hunk 8]
```diff
@@ -0,0 +1,471 @@
...
+    " * Plot the 2m temperature over Switzerland\n",
+    " * Use the projection=ccrs.Robinson() (don't forget to transform the data)\n",
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:282]:
```suggestion
    " * Use the `projection=ccrs.Robinson()` (don't forget to transform the data)\n",
```

File: Part3/ex3_03_psyplot.ipynb [hunk 9]
```diff
@@ -0,0 +1,471 @@
...
+   "metadata": {},
+   "source": [
+    "### Exercise\n",
+    " * Plot the 2m temperature over Switzerland\n",
+    " * Use the projection=ccrs.Robinson() (don't forget to transform the data)\n",
+    " * Add the borders of Switzerland\n",
+    " * Cut off data outside of Switzerland (5.8<lon<10.7, 45.5<lat<48)\n",
+    " * Add a colorbar\n",
+    " * Set limits to the colorbar\n",
+    " * Use the colormap cmc.nuuk"
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:158]:
```suggestion
    " * Use the colormap `cmc.nuuk`"
```

File: Part3/ex3_03_psyplot.ipynb [hunk 10]
```diff
@@ -0,0 +1,471 @@
...
+    "Checkout the [available formatoptions](https://psyplot.github.io/psy-maps/generated/psyplot.project.plot.mapplot.html) and try to make the same plot as before.\n",
+    "\n",
+    "\n",
+    "### Exercise\n",
+    " Use plot_icon.update(..) to adapt the plot in the following way: \n",
+    " * Plot the 2m temperature over Switzerland\n",
+    " * Use the projection=ccrs.Robinson() (don't forget to transform the data)\n",
+    " * Cut off data outside of Switzerland (5.8<lon<10.7, 45.5<lat<48)\n",
+    " * Set limits to the colorbar\n",
+    " * Use the colormap cmc.nuuk\n",
```
Comment (mathause) [inline Part3/ex3_03_psyplot.ipynb:285]:
```suggestion
    " * Use the colormap `cmc.nuuk`\n",
```

File: Part4/ex4_4_cosmo_pollen_S_Adamov.ipynb [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "00008092",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ftp_pyvis = \"ftp://iacftp.ethz.ch/pub_read/alauber/pyvis/\"\n",
+    "filename = \"cosmo_pollen.nc\"\n",
+    "urllib.request.urlretrieve(ftp_pyvis+filename, filename)\n",
+    "ds_alnu = psy.open_dataset(\"cosmo_pollen.nc\")"
```
Comment (mathause) [inline Part4/ex4_4_cosmo_pollen_S_Adamov.ipynb:56]:
```suggestion
    "ftp_pyvis = \"ftp://iacftp.ethz.ch/pub_read/alauber/pyvis/\"\n",
    "filename = \"../data/cosmo_pollen.nc\"\n",
    "urllib.request.urlretrieve(ftp_pyvis + filename, filename)\n",
    "ds_alnu = psy.open_dataset(filename)"
```

**Comment by AnnikaLau** [comment]
@mathause : everything should be fixed now, so it should be ready to merge if you agree?
I can look into pooch later.

**Pull request merged**